lib/saml2.coffee

```diff
@@ -89,13 +106,25 @@ format_pem = (key, type) ->
 
 # Takes a compressed/base64 enoded @saml_request and @private_key and signs the request using RSA-SHA256. It returns
 # the result as an object containing the query parameters.
-sign_get_request = (saml_request, private_key) ->
-  data = "SAMLRequest=" + encodeURIComponent(saml_request) + "&SigAlg=" + encodeURIComponent('http://www.w3.org/2001/04/xmldsig-more#rsa-sha256')
+sign_get_request = (saml_request, private_key, relay_state, response=false) ->
+  action = if response then "SAMLResponse" else "SAMLRequest"
+  data = "#{action}=" + encodeURIComponent(saml_request)
+  if relay_state
+    data += "&RelayState=" + encodeURIComponent(relay_state)
+  data += "&SigAlg=" + encodeURIComponent('http://www.w3.org/2001/04/xmldsig-more#rsa-sha256')
```

Comment:
Can we abstract this out then? A version of as.stringify that takes an array of key value pairs instead of an object for example. I don't like this manual construction littering up the code, it seems like it's easy to misunderstand or make a mistake with (forget an & for example). It also makes the code hard to parse.